pkg/ingestor/collector/collector.go

```diff
@@ -15,4 +15,77 @@
 
 package collector
 
-type Collector interface{}
+import (
+	"context"
+
+	"github.com/guacsec/guac/pkg/ingestor/processor"
+	"github.com/sirupsen/logrus"
+)
+
+type Collector interface {
+	// Is the collector enabled
+	Enabled(ctx context.Context) bool
+	// Retrieve the artifacts from the collector
+	RetrieveArtifacts(ctx context.Context) ([]*processor.Document, error)
+	// Indicated when the collector is done
+	IsDone() bool
+	// Type of the collector
+	Type() CollectorType
+}
+
+type CollectorType string
+
+const (
+	CollectorGCS    = "GCS"
+	CollectorOCI    = "OCI"
+	CollectorPubSub = "PubSub"
+	CollectorRekor  = "Rekor"
+)
+
+var (
+	documentCollector = map[CollectorType]Collector{}
+)
+
+// RegisterDocumentCollector is used to register new collector types
+func RegisterDocumentCollector(p Collector, d CollectorType) {
+	if _, ok := documentCollector[d]; ok {
+		logrus.Warnf("the document processor is being overwritten: %s", d)
+	}
+	documentCollector[d] = p
+}
+
+// Enabled checks to see if the collector type is properly initialized
+func Enabled(ctx context.Context) bool {
```

Comment:
this probably shouldn't be an function of the collector interface since it relies on the construct that consumes it, making there an circular dependency (i.e. if we have a separate package for each collector it probably will complain)

Reply:
removed.